ehrql/dummy_data_nextgen/generator.py

```diff
@@ -202,23 +207,31 @@ def generate_patient_facts(self, patient_id):
                 )
 
             dod_column = self.get_patient_column("date_of_death")
-            if dod_column is not None and dod_column.get_constraint(
-                Constraint.GeneralRange
-            ):
+            if dod_column is not None:
                 date_of_death = self.get_random_value(dod_column)
             else:
                 age_days = self.rnd.randrange(105 * 365)
                 date_of_death = date_of_birth + timedelta(days=age_days)
 
-            if date_of_death >= date_of_birth and (
-                date_of_death - date_of_birth < timedelta(105 * 365)
+            if (
+                date_of_birth is None
```

Comment:
I think so yeah. It can't with the real `patients` tableschema but we've got a number of other definitions of a patients table in the tests that don't have a not null constraint on date of birth.